File: c/api/WebhooksAPI.c
#include <stdlib.h>
#include <stdio.h>
#include <ctype.h>
#include "WebhooksAPI.h"

#define MAX_NUMBER_LENGTH 16
#define MAX_BUFFER_LENGTH 4096
#define intToStr(dst, src) \
    do {\
    char dst[256];\
    snprintf(dst, 256, "%ld", (long int)(src));\
}while(0)


// Outgoing webhooks
//
// Outgoing webhooks allows to build or set up Zulip integrations which are notified when certain types of messages are sent in Zulip. 
//
inline_response_200_t*
WebhooksAPI_zulipOutgoingWebhooks(apiClient_t *apiClient)
{
    list_t    *localVarQueryParameters = NULL;
    list_t    *localVarHeaderParameters = NULL;
    list_t    *localVarFormParameters = NULL;
    list_t *localVarHeaderType = list_create();
    list_t *localVarContentType = NULL;
    char      *localVarBodyParameters = NULL;

    // create the path
    long sizeOfPath = strlen("/zulip-outgoing-webhook")+1;
    char *localVarPath = malloc(sizeOfPath);
    snprintf(localVarPath, sizeOfPath, "/zulip-outgoing-webhook");



    list_addElement(localVarHeaderType,"application/json"); //produces
    apiClient_invoke(apiClient,
                    localVarPath,
                    localVarQueryParameters,
                    localVarHeaderParameters,
                    localVarFormParameters,
                    localVarHeaderType,
                    localVarContentType,
                    localVarBodyParameters,
                    "POST");

    if (apiClient->response_code == 200) {
        printf("%s\n","Success ");
    }
    //nonprimitive not container
    cJSON *WebhooksAPIlocalVarJSON = cJSON_Parse(apiClient->dataReceived);
    inline_response_200_t *elementToReturn = inline_response_200_parseFromJSON(WebhooksAPIlocalVarJSON);
    cJSON_Delete(WebhooksAPIlocalVarJSON);
    if(elementToReturn == NULL) {
        // return 0;
    }

    //return type
    if (apiClient->dataReceived) {
        free(apiClient->dataReceived);
        apiClient->dataReceived = NULL;
        apiClient->dataReceivedLen = 0;
    }
    
    
    
    list_free(localVarHeaderType);
    
    free(localVarPath);
    return elementToReturn;
end:
    free(localVarPath);
    return NULL;

}

File: dart-jaguar/lib/api/webhooks_api.dart
import 'package:jaguar_retrofit/annotations/annotations.dart';
import 'package:jaguar_retrofit/jaguar_retrofit.dart';
import 'package:jaguar_serializer/jaguar_serializer.dart';
import 'package:jaguar_mimetype/jaguar_mimetype.dart';
import 'dart:async';

import 'package:openapi/model/inline_response200.dart';

part 'webhooks_api.jretro.dart';

@GenApiClient()
class WebhooksApi extends ApiClient with _$WebhooksApiClient {
    final Route base;
    final Map<String, CodecRepo> converters;
    final Duration timeout;

    WebhooksApi({this.base, this.converters, this.timeout = const Duration(minutes: 2)});

    /// Outgoing webhooks
    ///
    /// Outgoing webhooks allows to build or set up Zulip integrations which are notified when certain types of messages are sent in Zulip. 
    @PostReq(path: "/zulip-outgoing-webhook")
    Future<InlineResponse200> zulipOutgoingWebhooks(
        ) {
        return super.zulipOutgoingWebhooks(

        ).timeout(timeout);
    }


}
